Userland/Libraries/LibDisassembly/riscv64/FD.cpp

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2023, kleines Filmröllchen <filmroellchen@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "FD.h"
+
+namespace Disassembly::RISCV64 {
+
+NonnullOwnPtr<InstructionImpl> parse_op_fp(u32 instruction)
+{
+    auto raw_parts = RawRType::parse(instruction);
+    auto width = static_cast<FloatWidth>(raw_parts.funct7 & 0b11);
+    auto rounding_mode = static_cast<RoundingMode>(raw_parts.funct3);
+    // Not mentioned in the spec, but bit 3 of funct7 distinguishes between float-to-integer (0) and integer-to-float (1) conversions in relevant instructions.
+    auto is_int_to_float = (raw_parts.funct7 & (1 << 3)) > 0;
+
+    auto operation = FloatArithmeticInstruction::Operation::Add;
+    switch (raw_parts.funct7 & ~0b11) {
+    case 0b0000000:
```

Comment:
Same comment as in `A`

Reply:
as discussed, won’t yield significant results